### ISSUE TITLE
SC1036 error on valid Bash function name

### ISSUE THREAD

**Issue by fidian** [koalaman/shellcheck#949]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC1036
- My shellcheck version (`shellcheck --version` or "online"): 0.3.7
- [X] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/usr/bin/env bash
@func() { echo "Hello"; }
@func
```

#### Here's what shellcheck currently says:

```
Line 2:
@func() { echo "Hello"; }
     ^-- SC1036: '(' is invalid here. Did you forget to escape it?
     ^-- SC1088: Parsing stopped here. Invalid use of parentheses?
```

#### Here's what I wanted or expected to see:

No errors.

I've tested this against Bash 3.2.48 and 4.3.48. Both handle the script just fine. I don't believe that the parenthesis can be escaped, nor would I want that because then potentially the parenthesis would be part of the function name. I did try escaping a parenthesis just for fun and that produced different errors and didn't work in Bash.
